src/commands/Games/higherLower.ts

```diff
@@ -0,0 +1,251 @@
+
+		await this.newRound(this.random(), message, undefined, message.channel.id, message.author.id, wager);
+	}
+
+	private reactionHandler(reaction: LLRCData) {
+		// If the reaction comes from herself... just dont
+		if (reaction.userID === this.client.user!.id) return;
+		// If the emoji reacted doesnt match our emojiset, just... don't
+		if (!(Object.values(ReactionEmoji) as string[]).includes(reaction.emoji.name)) return;
+		// If the message reacted to didn't correspond to a game's message, abort
+		const gameMessage = this.games.find(g => g.gameMessage.split('.')[0] === reaction.channel.id);
+		if (!gameMessage) return;
+		// If the user that reacted to the message doesn't have a game, abort
+		const gameData = this.games.get(reaction.userID);
+		if (!gameData) return;
+		// If the reaction user isnt the game user, ignore
+		if (reaction.userID !== gameData.user) return;
+		return this.gameHandler(reaction, gameData);
+	}
+
+	private async gameHandler(reaction: LLRCData, gameData: HigherLowerGameData) {
+		const gameMessage = this.getGameMessage(gameData.gameMessage);
+		await gameMessage.reactions.removeAll();
+		gameData.llrc.end();
+
+		const winningNumber = this.random();
+		switch (reaction.emoji.name) {
+			case ReactionEmoji.HIGHER:
+				if (winningNumber > gameData.previousNumber) return this.win(gameData, winningNumber);
+				return this.loss(gameData, winningNumber);
+
+			case ReactionEmoji.LOWER:
+				if (winningNumber < gameData.previousNumber) return this.win(gameData, winningNumber);
+				return this.loss(gameData, winningNumber);
+
+			case ReactionEmoji.CASHOUT:
+				await this.cashout(gameData, gameMessage);
+				break;
+
+			case ReactionEmoji.CANCEL:
+				this.games.delete(gameData.user);
+				await gameMessage.edit('Game cancelled. See you again next time!');
+				break;
+			default:
+				break;
+		}
+	}
+
+	private async newRound(previousNumber: number, message: Message, previousGameData?: HigherLowerGameData, channelID?: string, userID?: string, wagerIn?: number) {
+		// Maybe use a function for this?
+		const channel = typeof channelID === 'undefined'
+			? typeof previousGameData === 'undefined'
+				? null
+				: this.client.channels.get(previousGameData.gameMessage.split('.')[0]) as TextChannel
+			: this.client.channels.get(channelID) as TextChannel;
+		if (channel === null || channel === undefined) throw new Error('No channel provided');
+
+		const user = typeof userID === 'undefined'
+			? typeof previousGameData === 'undefined'
+				? null
+				: previousGameData.user
+			: userID;
+		if (user === null) throw new Error('No user provided');
+
+		const wager = typeof wagerIn === 'undefined'
+			? typeof previousGameData === 'undefined'
+				? null
+				: previousGameData.wager
+			: wagerIn;
+		if (wager === null) throw new Error('No wager provided');
+
+		const turn = typeof previousGameData === 'undefined'
+			? 1
+			: previousGameData.turn + 1;
```

Comment:
What in the devil is this? Just make a clean object, you can take inspiration on how HungerGames does it.

Reply:
I could compact this as an object... but... I don't think you'd like it.
I cannot use HG's implementation because
a) It only has 2 values. truthy or falsy. Here we have 3, and they aren't even standardized under a specific type
b) you do some weird finicking around with promises I am unable to comprehend. You are... resolving a promise from another function and whatnot